earth2grid/healpix.py

```diff
@@ -602,3 +599,154 @@ def conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1):
     weight = weight.unsqueeze(-3)
     out = torch.nn.functional.conv3d(input, weight, bias, stride, padding, dilation, groups)
     return einops.rearrange(out, "n c f x y -> n c () (f x y)")
+
+
+def _pixels_to_rings(nside: int, p: ArrayT) -> ArrayT:
+    """Get the ring number of a pixel ``i`` in RING order"""
+    # See eq (2-5) of Gorski
+    xp = _get_array_library(p)
+    npix = 12 * nside * nside
+    ncap = 2 * nside * (nside - 1)
+
+    i_north = xp.floor(0.5 * (1 + np.sqrt(1 + 2 * p)))
```

Comment:
Not for this PR since torch doesn't compy, but could potentially replace this `_get_array_library` pattern with the Array API [here](https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__floordiv__.html#array_api.array.__floordiv__) and other similar functions.

Reply:
Cool! I didn't realize that existed. I wonder if/when pytorch will adopt it.